### PR TITLE
Add doc to setup continuous integration E2E tests for extensions

### PR DESCRIPTION
**Pull request by gtardif**
### Proposed changes

Add documentation explaining how to setup E2E tests for extensions in CI.

### CODE REVIEW

File: desktop/extensions-sdk/dev/continuous-integration.md [hunk 1]
```diff
@@ -0,0 +1,78 @@
+---
+title: "Continuous Integration"
+description: Automatically test and validate your extension.
+keywords: Docker, Extensions, sdk, CI, test, regression
+---
+
+In order to help validating your extension and ensure it's functional, Docker Desktop provides a set of tools to help you set continuous integration for your extension.
```
Comment (benja-M-1) [inline desktop/extensions-sdk/dev/continuous-integration.md:7]: Maybe we should mention that they are experimental, as we did in the package readme. WDYT @gtardif?

File: desktop/extensions-sdk/dev/continuous-integration.md [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+You need Docker Desktop to be able to install and validate your extension.
+You can start Docker Desktop in Github Actions using the [Docker Desktop Action](https://github.com/docker/desktop-action), with the following step:
+
+```yaml
+steps:
+  - id: start_desktop
+    uses: docker/desktop-action/start@v0.1.0
+```
+
+> This action supports only Github Action Mac nodes at the moment ; you need to specify `runs-on: macOS-latest` for your end to end tests.
```
Comment (benja-M-1) [inline desktop/extensions-sdk/dev/continuous-integration.md:20]: In the GHA doc they call nodes runners: https://docs.github.com/en/actions/learn-github-actions/understanding-github-actions#runners

```suggestion
> This action supports only Github Action macOS runners at the moment ; you need to specify `runs-on: macOS-latest` for your end to end tests.
```

**Review by benja-M-1 — CHANGES_REQUESTED** [review]
Minore change request about GitHub Actions lingo

**Review by benja-M-1 — APPROVED** [review]

File: desktop/extensions-sdk/dev/continuous-integration.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
+---
+title: "Continuous Integration"
+description: Automatically test and validate your extension.
+keywords: Docker, Extensions, sdk, CI, test, regression
+---
+
+In order to help validating your extension and ensure it's functional, the Extension SDK provides a set of tools to help you set continuous integration for your extension.
+
+> The github action and the extension-test-helper library are both [experimental](https://docs.docker.com/release-lifecycle/#experimental).
```
Comment (ChrisChinchilla) [inline desktop/extensions-sdk/dev/continuous-integration.md:9]: Is there a link for more info on the extension-test-helper library?

File: desktop/extensions-sdk/dev/continuous-integration.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
+---
+title: "Continuous Integration"
```
Comment (ChrisChinchilla) [inline desktop/extensions-sdk/dev/continuous-integration.md:2]: I know it's obvious in this context but just so we confirm to style and unpack the acronym, how about…

```suggestion
title: "Continuous Integration (CI)"
```

File: desktop/extensions-sdk/dev/continuous-integration.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+You need Docker Desktop to be able to install and validate your extension.
+You can start Docker Desktop in Github Actions using the [Docker Desktop Action](https://github.com/docker/desktop-action), with the following step:
+
+```yaml
+steps:
+  - id: start_desktop
+    uses: docker/desktop-action/start@v0.1.0
+```
+
+> This action supports only Github Action macOS runners at the moment ; you need to specify `runs-on: macOS-latest` for your end to end tests.
```
Comment (ChrisChinchilla) [inline desktop/extensions-sdk/dev/continuous-integration.md:24]:
```suggestion
> This action supports only Github Action macOS runners at the moment. You need to specify `runs-on: macOS-latest` for your end to end tests.
```

File: desktop/extensions-sdk/dev/continuous-integration.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+{: .important }
+
+Once the step has executed, the next steps can use Docker Desktop and the Docker CLI to install an test the extension
```
Comment (ChrisChinchilla) [inline desktop/extensions-sdk/dev/continuous-integration.md:27]:
```suggestion
Once the step has executed, the next steps can use Docker Desktop and the Docker CLI to install and test the extension.
```

File: desktop/extensions-sdk/dev/continuous-integration.md [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+
+## Validating your extension with puppeteer
+
+Once Docker Desktop is started in the CI, you can build, install and validate your extension with jest and puppeteer.
```
Comment (ChrisChinchilla) [inline desktop/extensions-sdk/dev/continuous-integration.md:31]:
```suggestion
Once Docker Desktop is started in the CI, you can build, install, and validate your extension with jest and puppeteer.
```

**Review by ChrisChinchilla — CHANGES_REQUESTED** [review]
Some small typo and style suggestions, but thanks for the contribution!

File: desktop/extensions-sdk/dev/continuous-integration.md (hunk 3 above)
Comment (gtardif) [reply desktop/extensions-sdk/dev/continuous-integration.md]: Good catch, I added both links here (the GHA link was already mentioned below)

**Review by ChrisChinchilla — DISMISSED** [review]

**Comment by gtardif** [comment]
@ChrisChinchilla can you merge this PR ? (or someone with merge rights)

**Comment by ChrisChinchilla** [comment]
@gtardif Sure, as this is quite a big change I am just going to pull a local copy and run a few linters etc

**Review by ChrisChinchilla — APPROVED** [review]

**Pull request merged**